### PR TITLE
Allow accredited providers to publish courses

### PR DESCRIPTION
**Pull request by JR-G**
### Context

We need to tweak earlier validations as accredited providers who publish courses will have an accredited provider value as ‘nil’ on the course. 

We should not prevent publication when the value is nil - in fact this should be the only situation where this happens.

image

### Changes proposed in this pull request

Do not check for the presence of `accrediting_provider` on a `course` if the provider is accredited.

### Checklist

- [ ] Make sure all information from the Trello card is in here
- [ ] Attach to Trello card
- [ ] Rebased main
- [ ] Cleaned commit history
- [ ] Tested by running locally
- [ ] Inform data insights team due to database changes

### CODE REVIEW

**Review by avinhurry — APPROVED** [review]
I was thinking we will probably have quite a few edge cases like this from the provider reclassification. 

It might be worth making a folder for these. "provider_reclassification" or something. WDYT? It might be difficult to know what the test is doing as "publishing_a_course_as_an_accredited_provider_spec" feels generic. 

Happy to do as a follow up though considering the impact this is having on the user

File: spec/features/publish/courses/publishing_a_course_as_an_accredited_provider_spec.rb [hunk 1]
```diff
@@ -0,0 +1,56 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+feature 'Publishing a course when course when accrediting_provider is nil', { can_edit_current_and_next_cycles: false } do
```
[the remark this answers is not in the document]
Comment (JR-G) [reply spec/features/publish/courses/publishing_a_course_as_an_accredited_provider_spec.rb:5]: yeah it's a fair point – I'm quite keen to get this in asap for the providers, but the tests and scenarios will/do need to be properly addressed
